### ISSUE TITLE
Question : How to use with navkey ?

### ISSUE THREAD

**Issue by bounty1342** [Milad-Akarie/auto_route_library#45]
Hi,

I have trouble using this package with navKey. Do you have an example by any chance ?

I have declared :
`static final navKey = new GlobalKey<ExtendedNavigatorState<Router>>();`
but when I access  `navKey.currentState`is always null.

Regards,

**Comment by bounty1342** [comment]
You should add it to the documentation, I was still adding it to the MaterialApp.
``` dart
builder: ExtendedNavigator<Router>(
                router: Router(),
                key: MyNavigationService.navKey,
                observers: [HeroController()],
              )
```

@ResoCoder : might be good to update the tutorial a bit, most is still valid.

**Comment by Milad-Akarie** [comment]
Hey, @bounty1342 I'm glad you're liking the package. 
passing your own key isn't required anymore, you can easily obtain the current navigation state  without context using
`ExtendedNavigatore.ofRouter<Router>()`
It was mentioned in the updated README file

**Comment by bounty1342** [comment]
My service was using navKey, but I changed it.

Only remaining issue is with onWillPop that is not triggered anymore, any recommendation ?

 ``` dart
@override
  Widget build(BuildContext context) {
    return WillPopScope(
      onWillPop: () => _exitApp(context),
```

 ``` dart
    NavigationService navigationProvider =
        Provider.of<NavigationService>(context, listen: false);
    if (!navigationProvider.hasNoRoute()) {
      navigationProvider.navigationBack();
      return Future.value(false);
    }

    return showDialog(
        context: context,
        builder: (BuildContext context) {
          return AlertDialog(
            title: new Text(I18n.of(context).exitApplicationQuestionTitle),
            content: new Text(I18n.of(context).exitApplicationQuestionLbl),
            actions: <Widget>[
              FlatButton(
                onPressed: () => Navigator.of(context).pop(false),
                child: new Text(I18n.of(context).exitApplicationQuestionNoBtn),
              ),
              FlatButton(
                onPressed: () => Navigator.of(context).pop(true),
                child: new Text(I18n.of(context).exitApplicationQuestionYesBtn),
              ),
            ],
          );
        });
  }
```

**Comment by Milad-Akarie** [comment]
@bounty1342  can I see your MaterialApp setup? 
also may I ask, why do you need a navigation service if you're gonna Provide it with context? 
why not use ExtendedNavigator.of(context)? the ExtendedNavigator state is already provided in the context

**Comment by bounty1342** [comment]
@Milad-Akarie,

My MaterialApp setup below :
```
 child: MaterialApp(
              localizationsDelegates: [
                _localeOverrideDelegate,
                const I18nDelegate(),
                GlobalMaterialLocalizations.delegate,
                GlobalWidgetsLocalizations.delegate,
                GlobalCupertinoLocalizations.delegate,
              ],
              supportedLocales: MyI18nDelegate.supportedLocals,
              debugShowCheckedModeBanner: appConfig.debugShowCheckedModeBanner,
              title: appConfig.appName,
              theme: themeService.getTheme(),
              darkTheme: themeService.getThemeDark(),
              builder: ExtendedNavigator<Router>(
                router: Router(),
              )),
```

I use a MVVM pattern, the services know nothing about the context. This is to enforce between the logic and the UI.

issue #47, seems to be related.

Regards,

**Comment by Milad-Akarie** [comment]
@bounty1342  should be fixed in auto_route: 0.4.2, check it out.
